renku/core/util/metadata.py

```diff
@@ -42,7 +42,7 @@ def construct_creators(creators: List[Union[dict, str]], ignore_email=False) ->
     """Parse input and return a list of Person."""
     creators = creators or []
 
-    if not isinstance(creators, Iterable) or isinstance(creators, str):
+    if not isinstance(creators, Iterable):
```

Comment:
Why is this removed? Here we except a list of creators and a string would mean a single creator.

Reply:
That part is a bit annoying, mypy will complain about dead code, since as per the type signature, this code is nonsensical. Actually the whole `if` would be, I'm not sure why it doesn't complain about the first part.

But of course in python you could pass whatever since the types aren't enforced, and mypy will pretend they're enforced (since they are for our code, by mypy) and complain

This discrepancy is what led me to open/write https://github.com/SwissDataScienceCenter/renku-python/issues/3188 , which got completed first. We don't want that everywhere for performance reasons, but having it around our core layer to protect from the outside world makes sense to me. So I think the whole `if` above could be removed now.